node-repository/src/main/java/com/yahoo/vespa/hosted/provision/Node.java

```diff
@@ -103,9 +103,9 @@ public Node(String id, IP.Config ipConfig, String hostname, Optional<String> par
         }
 
         if (type != NodeType.host && reservedTo.isPresent())
-            throw new IllegalArgumentException("Only hosts can be reserved to a tenant");
+            throw new IllegalArgumentException("Only tenant hosts can be reserved to a tenant");
 
-        if (type != NodeType.host && exclusiveTo.isPresent())
+        if (!type.isHost() && exclusiveTo.isPresent())
```

Comment:
Any reason to allow this for other hosts?

Reply:
Yes, infra applications are `exclusive` which causes the `HostProvisioner` implemention to set `exclusiveTo` for those hosts.